spec/src/language_negotiation_lambda_spec.js

```diff
@@ -0,0 +1,258 @@
+            { header: "EN-US",                customHeader: "en" },
+            { header: "ES-419",               customHeader: "es" }
+        ].forEach( function( headers ) {
+            it( `Resolves header ${headers.header} to ${headers.customHeader}`, function( done ) {
+                event.Records[0].cf.request.headers['Accept-Language'] = [headers.header];
+
+                languageNegotiationLambda.handler( event, { /* context */ }, (err, result) => {
+                    try {
+                        expect( err ).toBeNull();
+                        expect( result.headers['X-Accept-Language'] ).toEqual([headers.customHeader]);
+                        done();
+                    }
+                    catch( error ) {
+                        done( error );
+                    }
+                });
+            });
+        });
+    });
+
+    describe( 'Malformed cookies', function(){
+        [
+            undefined,
+            NaN,
+            null
+        ].forEach( function( cookie ) {
+            it( `Passes on cookie ${cookie} and uses browser language`, function( done ) {
+                event.Records[0].cf.request.headers.Cookie = [cookie];
+                event.Records[0].cf.request.headers['Accept-Language'] = ['es'];
+                spyOn(console, "log");
+
+                languageNegotiationLambda.handler( event, { /* context */ }, (err, result) => {
+                    try {
+                        expect( err ).toBeNull();
+                        expect( result.headers['X-Accept-Language'] ).toEqual(['es']);
+                        expect( console.log ).toHaveBeenCalled();
+                        done();
+                    }
+                    catch( error ) {
+                        done( error );
+                    }
+                });
+            });
+        });
+    });
+
+    describe( 'Malformed headers', function(){
+        [
+            undefined,
+            NaN,
+            null
+        ].forEach( function( header ) {
+            it( `Resolves header ${header} to default and logs error`, function( done ) {
+                event.Records[0].cf.request.headers['Accept-Language'] = [header];
+                spyOn(console, "log");
+
+                languageNegotiationLambda.handler( event, { /* context */ }, (err, result) => {
+                    try {
+                        expect( err ).toBeNull();
+                        expect( result.headers['X-Accept-Language'] ).toEqual(default_header);
+                        expect( console.log ).toHaveBeenCalled();
+                        done();
+                    }
+                    catch( error ) {
+                        done( error );
+                    }
+                });
+            });
+        });
+
+        [
+            "",
+            "dhbeiyu292dfiue2"
+        ].forEach( function( header ) {
+            it( `Resolves header ${header} to default`, function( done ) {
```

Comment:
This seems like mostly the same test as line 156. Could we move these test cases there?